akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -299,6 +437,239 @@ abstract class GraphStageLogic private[stream] (inCount: Int, outCount: Int) {
    */
   final protected def isClosed[T](out: Outlet[T]): Boolean = (interpreter.portStates(conn(out)) & OutClosed) != 0
 
+  private val emptyArray = new ArrayBuffer[Any](0)
```

Comment:
I dislike the extra field. I removed quite a bit of memory overhead in the other PR, so I would not like to see it again :)